mlflow/pyfunc/__init__.py

```diff
@@ -2270,7 +2290,24 @@ def predict(model_input: List[str]) -> List[str]:
     """
     _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
     _validate_pyfunc_model_config(model_config)
+    _validate_and_prepare_target_save_path(path)
+
+    model_code_path = None
     if python_model:
+        if isinstance(model_config, Path):
+            model_config = os.fspath(model_config)
```

Comment:
@annzhang-db Do we have a test for the case where `model_config` is a `Path` instance? If not, can we add one?